packages/blockprotocol/cli/commands/publish/shared/find-project-root.js

```diff
@@ -0,0 +1,13 @@
+import { findUp } from "find-up";
+import path from "node:path";
+
+/**
+ * Find the path to the project root, by searching upwards for the nearest package.json
+ * @returns {Promise<string>}
```

Comment:
```suggestion
 * @returns {Promise<string|undefined>}
```

If we switch to search for `.bprc.json` via `find-up`, this function will be used only once, so won't be needed.

Reply:
https://github.com/blockprotocol/blockprotocol/pull/575/commits/6ca50aaf85446fc13c248004439d1929ff9261a0

It is still needed in the API key file in order to find the location in which to create one, if the user wants

Reply:
We could alternatively create it in their user directory. Having thought about it, we probably want to add it to their `.gitignore` as well if we create it in the project...